tests/components/demo/test_vacuum.py

```diff
@@ -47,326 +47,293 @@
+
+    await common.async_locate(hass, ENTITY_VACUUM_NONE)
+    state = hass.states.get(ENTITY_VACUUM_NONE)
+    assert state.attributes.get(ATTR_STATUS) is None
+
+    await common.async_return_to_base(hass, ENTITY_VACUUM_NONE)
+    state = hass.states.get(ENTITY_VACUUM_NONE)
+    assert state.attributes.get(ATTR_STATUS) is None
+
+    await common.async_set_fan_speed(hass, FAN_SPEEDS[-1], entity_id=ENTITY_VACUUM_NONE)
+    state = hass.states.get(ENTITY_VACUUM_NONE)
+    assert state.attributes.get(ATTR_FAN_SPEED) != FAN_SPEEDS[-1]
+
+    await common.async_clean_spot(hass, entity_id=ENTITY_VACUUM_BASIC)
+    state = hass.states.get(ENTITY_VACUUM_BASIC)
+    assert "spot" not in state.attributes.get(ATTR_STATUS)
+    assert state.state == STATE_OFF
+
+    # VacuumDevice should not support start and pause methods.
+    hass.states.async_set(ENTITY_VACUUM_COMPLETE, STATE_ON)
+    await hass.async_block_till_done()
+    assert vacuum.is_on(hass, ENTITY_VACUUM_COMPLETE)
+
+    await common.async_pause(hass, ENTITY_VACUUM_COMPLETE)
+    assert vacuum.is_on(hass, ENTITY_VACUUM_COMPLETE)
+
+    hass.states.async_set(ENTITY_VACUUM_COMPLETE, STATE_OFF)
+    await hass.async_block_till_done()
+    assert not vacuum.is_on(hass, ENTITY_VACUUM_COMPLETE)
+
+    await common.async_start(hass, ENTITY_VACUUM_COMPLETE)
+    assert not vacuum.is_on(hass, ENTITY_VACUUM_COMPLETE)
+
+    # StateVacuumDevice does not support on/off
+    await common.async_turn_on(hass, entity_id=ENTITY_VACUUM_STATE)
+    state = hass.states.get(ENTITY_VACUUM_STATE)
+    assert state.state != STATE_CLEANING
+
+    await common.async_turn_off(hass, entity_id=ENTITY_VACUUM_STATE)
+    state = hass.states.get(ENTITY_VACUUM_STATE)
+    assert state.state != STATE_RETURNING
+
+    await common.async_toggle(hass, entity_id=ENTITY_VACUUM_STATE)
+    state = hass.states.get(ENTITY_VACUUM_STATE)
+    assert state.state != STATE_CLEANING
+
+
+async def test_services(hass):
+    """Test vacuum services."""
+    # Test send_command
+    send_command_calls = async_mock_service(hass, DOMAIN, SERVICE_SEND_COMMAND)
+
+    params = {"rotate": 150, "speed": 20}
+    await common.async_send_command(
+        hass, "test_command", entity_id=ENTITY_VACUUM_BASIC, params=params
+    )
+    assert len(send_command_calls) == 1
+    call = send_command_calls[-1]
+
+    assert call.domain == DOMAIN
+    assert call.service == SERVICE_SEND_COMMAND
+    assert call.data[ATTR_ENTITY_ID] == ENTITY_VACUUM_BASIC
+    assert call.data[ATTR_COMMAND] == "test_command"
+    assert call.data[ATTR_PARAMS] == params
+
+    # Test set fan speed
+    set_fan_speed_calls = async_mock_service(hass, DOMAIN, SERVICE_SET_FAN_SPEED)
+
+    await common.async_set_fan_speed(
+        hass, FAN_SPEEDS[0], entity_id=ENTITY_VACUUM_COMPLETE
+    )
+    assert len(set_fan_speed_calls) == 1
+    call = set_fan_speed_calls[-1]
+
+    assert call.domain == DOMAIN
+    assert call.service == SERVICE_SET_FAN_SPEED
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5315232)